### ISSUE TITLE
RPC test output is too verbose and includes expected failures by default

### ISSUE THREAD

**Issue by daira** [zcash/zcash#2453]
For example, successful test output for wallet_protectcoinbase.py includes, by default:
```
  waiting for async operation opid-22ebcf29-7202-41af-8034-3399d0bd5fb9
  ...returned status: failed
  ...returned error: Insufficient transparent funds, have 10.00, need 0.00000053 more to avoid creating invalid change output 0.00000001 (dust threshold is 0.00000054)
  waiting for async operation opid-f0dacab2-989e-40bc-955e-29d35b2c8513
  ...returned status: failed
  ...returned error: Insufficient transparent funds, have 10.00, need 10000.0001
  waiting for async operation opid-a0ddd003-6a68-4bf3-9f2e-32db54bdf925
  ...returned status: failed
  ...returned error: Insufficient protected funds, have 9.9998, need 10000.0001
```

which could lead someone to conclude there is a problem when there is none.

**Comment by per-gron** [comment]
Note to others: I have started working on this. See https://github.com/per-gron/zcash/tree/less-verbose-tests

**Comment by daira** [comment]
Fixed by #2796.
